src/hits/hits.ts

```diff
@@ -35,30 +47,31 @@ import { NgAisInstantSearch } from '../instantsearch/instantsearch';
 export class NgAisHits extends BaseWidget {
   @ContentChild(TemplateRef) public template?: TemplateRef<any>;
 
-  // rendering options
-  @Input() transformItems?: Function;
+  @Input()
+  public transformItems?: <U extends HitsItem>(items: HitsItem[]) => U[];
 
-  // inner widget state returned from connector
-  public state: { hits: {}[]; results: {} } = { hits: [], results: {} };
+  public state: HitsState = {
+    hits: [],
+    results: {},
+  };
 
   constructor(
     @Inject(forwardRef(() => NgAisInstantSearch))
     public instantSearchParent: any
   ) {
     super('Hits');
-    this.createWidget(connectHitsWithInsights, { escapeHits: true });
+  }
+
+  ngOnInit() {
+    this.createWidget(connectHitsWithInsights, {
+      escapeHits: true,
+      transformItems: this.transformItems,
+    });
+    super.ngOnInit();
   }
 
   updateState = (state, isFirstRendering: boolean) => {
     if (isFirstRendering) return;
-
-    this.state = {
-      ...state,
-      results: state.results,
-      hits:
-        typeof this.transformItems === 'function'
-          ? this.transformItems(state.hits)
-          : state.hits,
-    };
+    this.state = state;
```

Comment:
I have not idea why this was written like this in the first place.
I was suspecting cases where the widget didn't refresh properly but this never happened in my testing.

Reply:
I'm pretty sure that was to make sure the transform items applied